hudi-client/hudi-client-common/src/main/java/org/apache/hudi/client/transaction/lock/audit/StorageLockProviderAuditService.java

```diff
@@ -0,0 +1,139 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.hudi.client.transaction.lock.audit;
+
+import org.apache.hudi.client.transaction.lock.StorageLockClient;
+import org.apache.hudi.common.table.HoodieTableMetaClient;
+import org.apache.hudi.storage.StoragePath;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.HashMap;
+import java.util.Map;
+import java.util.function.Function;
+import java.util.function.Supplier;
+
+/**
+ * Storage-based audit service implementation for lock provider operations.
+ * Writes audit records to a single JSONL file per transaction to track lock lifecycle events.
+ */
+public class StorageLockProviderAuditService implements AuditService {
+  
+  private static final Logger LOG = LoggerFactory.getLogger(StorageLockProviderAuditService.class);
+  private static final ObjectMapper OBJECT_MAPPER = new ObjectMapper();
+  public static final String AUDIT_FOLDER_NAME = "audit";
+  
+  private final String ownerId;
+  private final long transactionStartTime;
+  private final String auditFilePath;
+  private final StorageLockClient storageLockClient;
+  private final Function<Long, Long> lockExpirationFunction;
+  private final Supplier<Boolean> lockHeldSupplier;
+  private final StringBuilder auditBuffer;
+  
+  /**
+   * Constructor for StorageLockProviderAuditService.
+   * 
+   * @param basePath The base path where audit files will be written
+   * @param ownerId The full owner ID
+   * @param transactionStartTime The timestamp when the transaction started (lock acquired)
+   * @param storageLockClient The storage client for writing audit files
+   * @param lockExpirationFunction Function that takes a timestamp and returns the lock expiration time
+   * @param lockHeldSupplier Supplier that provides whether the lock is currently held
+   */
+  public StorageLockProviderAuditService(
```

Comment:
to confirm, the audit service is created once every transaction, so we do not need to rewrite files?